driver-sync/src/main/com/mongodb/client/internal/CryptConnection.java

```diff
@@ -87,13 +93,19 @@ public ConnectionDescription getDescription() {
     @Override
     public <T> T command(final String database, final BsonDocument command, final FieldNameValidator commandFieldNameValidator,
             @Nullable final ReadPreference readPreference, final Decoder<T> commandResultDecoder,
-            final OperationContext operationContext, final boolean responseExpected,
-            @Nullable final SplittablePayload payload, @Nullable final FieldNameValidator payloadFieldNameValidator) {
+            final OperationContext operationContext, final boolean responseExpected, final OpMsgSequences sequences) {
 
         if (serverIsLessThanVersionFourDotTwo(wrapped.getDescription())) {
             throw new MongoClientException("Auto-encryption requires a minimum MongoDB version of 4.2");
         }
 
+        SplittablePayload payload = null;
+        FieldNameValidator payloadFieldNameValidator = null;
+        if (sequences instanceof ValidatableSplittablePayload) {
```

Comment:
Done in ffd4f75e3bba4060c578e009e0b34d9e1d35319b. I added assertion instead of throwing `MongoClientException`: `CryptConnection` cannot observe anything but `ValidatableSplittablePayload`, `EmptyMessageSequences` because of the `isTrue("`autoEncryptionSettings` is null, as bulkWrite does not currently support automatic encryption", autoEncryptionSettings == null)` check in `MongoClusterImpl`.